### PR TITLE
refactor(v2): refactor dark toggle into a hook

### PR DESCRIPTION
**Pull request by wgao19**
## Motivation

Refactor dark toggle logic into a hook

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

test on preview that:
- dark toggle still works
- sets theme in local storage when toggling
- navigating and refreshing page picks up the theme properly

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-classic/src/theme/Navbar/index.js [hunk 1]
```diff
@@ -17,6 +17,8 @@ import SearchBar from '@theme/SearchBar';
 
 import classnames from 'classnames';
 
+import useTheme from '../../hooks/theme';
```
Comment (endiliey) [inline packages/docusaurus-theme-classic/src/theme/Navbar/index.js]: should be @theme/hooks/theme

File: packages/docusaurus-theme-classic/src/hooks/theme.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import * as React from 'react';
+
+const useTheme = defaultTheme => {
+  const [theme, setTheme] = React.useState(defaultTheme);
```
Comment (endiliey) [inline packages/docusaurus-theme-classic/src/hooks/theme.js]: should we use 

```js
const defaultTheme = typeof document !== 'undefined'
      ? document.querySelector('html').getAttribute('data-theme')	 : ''
```

if its undefined ?

File: packages/docusaurus-theme-classic/src/theme/Navbar/index.js (hunk 1 above)
Comment (wgao19) [reply packages/docusaurus-theme-classic/src/theme/Navbar/index.js]: this means I should put hooks under `theme` directory is it?

Comment (endiliey) [reply packages/docusaurus-theme-classic/src/theme/Navbar/index.js]:
```suggestion
import useTheme from '@theme/hooks/theme';
```

iirc this works as well

**Comment by wgao19** [comment]
I'm reading the code from react day picker, it seems that the mutation observer is used to watch theme changes on the document element, whereas in our current implementation we're rendering that element with our `useTheme` returned `theme`. So looks like not a same mindset.

File: packages/docusaurus-theme-classic/src/hooks/theme.js (hunk 2 above)
Comment (wgao19) [reply packages/docusaurus-theme-classic/src/hooks/theme.js]: Updated to let the hook computes default by itself.

File: packages/docusaurus-theme-classic/src/theme/Navbar/index.js (hunk 1 above)
Comment (wgao19) [reply packages/docusaurus-theme-classic/src/theme/Navbar/index.js]: Moved hooks into `theme/`, not sure if this is intended or not?

**Pull request merged**